### PR TITLE
Add peaks=true to all audio searches

### LINKED ISSUE #1800: Include `peaks=true` param on audio requests
## Problem

In WordPress/openverse-api#917 we are changing the peaks of the waveform to be optional in API responses, so we need to adjust petitions here to ensure the Frontend keeps using real waveforms (when available) for audio tracks.

## Alternatives

The audio track component will fall back to using randomized deterministic peaks. This is less preferable as it does not give the user exact information on what type of audio can be found before playing it.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by zackkrida**
## Fixes

Fixes #1800 by @krysal 

## Description

Wow, I struggled to work with the media services. I cannot wait to refactor some of that code.

Anyway, this PR adds a special case to add `peaks=true` to all audio and related audio searches. To do so I also had to upgrade the related media method to allow for query parameters. Maybe that will be a nice addition in the future.

I'm not sure about tests here. Do we have any to verify the API endpoint paths are correct? Otherwise, this is covered by the fact that the visual regression tests would change if the peak data isn't downloaded. Still, it seems like we need better test coverage around the API endpoints in general.

## Testing Instructions

Check on client-rendered searches that the `peaks=true` is present for all audio API search requests, and on related audio searches. Finally, make sure you see the same peaks on SSR searches.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [ ] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by zackkrida** [comment]
Thanks for the fix, @obulat!

File: src/data/media-service.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { decodeMediaData } from '~/utils/decode-media-data'
 import type { ApiQueryParams } from '~/utils/search-query-transform'
 import type { ApiService } from '~/data/api-service'
 import type { DetailFromMediaType, Media } from '~/models/media'
-import type { SupportedMediaType } from '~/constants/media'
+import { AUDIO, SupportedMediaType } from '~/constants/media'
```
Comment (dhruvkb) [inline src/data/media-service.ts:5]:
```suggestion
import { AUDIO, type SupportedMediaType } from '~/constants/media'
```

**Review by dhruvkb — APPROVED** [review]
LGTM, except for 1 change to meet Vue 3 + TS linting requirements. Not sure why the linting process did not flag that.

File: src/data/media-service.ts (hunk 1 above)
Comment (obulat) [reply src/data/media-service.ts]: I didn't know you can add `type` in front of the named import! Thank you, @dhruvkb!

**Review by obulat — APPROVED** [review]
Adds peaks to all requests, SSR and CSR!
And I agree with your remark about the services.

**Pull request merged**